rows.go

```diff
@@ -64,12 +65,24 @@ func (rows *mysqlRows) Columns() []string {
 	return columns
 }
 
+func (rows *mysqlRows) setFinish(f func()) {
```

Comment:
If we also consider the interface and function definition, we don't save anything here but introduce some extra overhead and complexity.